### ISSUE TITLE
Debian 10 keybindings appear to fail

### ISSUE THREAD

**Issue by pevsonic** [mzur/gnome-shell-wsmatrix#74]
I've done a new install of debian buster 10.1 and installed workspace matrix via the extensions website. This appears to give me Gnome 3.30.2 and wsmatrix v1.1.0. 

Using the default keybindings for move-to-workspace and switch-to-workspace which are the same as Ubuntu 18.04 (my other install which works fine) for some reason only the up and down bindings work not left and right (for both move and switch).

The behaviour is such that on my standard 8 wide x 2 high workspace, starting top left, "down" moves workspaces as if they were a linear sequence along the top row L->R and then along the bottom row L->R. The animations of windows shows windows moving as if one had gone down rather than to the right however.

One thing to note is that when you press ctrl-alt-left and ctrl-alt-right whilst you don't change workspace, the overview still appears.

To the extent of information detailed in the issue, this does sound a bit like #65 (which happens to be gnome 3.30.1) so I wonder if the same thing occurring? 

As far as I can tell, on both Ubuntu 18.04 (working) and Debian 10 (not working) the gnome extensions site installs the same plugin version so the main difference is that ubuntu 18.04 is using gnome 3.28.

Happy to help debug but I'm an embedded C guy so might need a nudge in the right direction!

**Comment by pevsonic** [comment]
Digging further, there's a related underlying problem with mutter which I think affects the animations not the key bindings though : 
https://gitlab.gnome.org/GNOME/mutter/issues/270

**Comment by ebeem** [comment]
I will download and install a fresh `debian buster 10.1` and test this issue.
Thanks for spending time providing details.

**Comment by ebeem** [comment]
I could reproduce, will investigate more when I have time.

**Comment by mzur** [comment]
After re-reading your comment, I think the bug in Mutter is the cause of the keybindings not working. This has been reported a while back ( #4 ). The animation issue might be adressed by #70.

**Comment by ebeem** [comment]
I am not sure about the keybindings issue, but I know mutter 3.30 release had this animation issue where right/left animations were not properly displayed as @pevsonic mentioned in https://github.com/mzur/gnome-shell-wsmatrix/issues/74#issuecomment-553387566.

**Comment by mzur** [comment]
@pevsonic Can you confirm that an update/patch of Mutter fixes this issue for you?

**Comment by ebeem** [comment]
here's some debugging info for a 3x3 grid, the following graph indicates the index number of each workspace starting from 0, first workspace (top-left) until 8, last workspace (bottom-right)

```
0    1    2
3   [4]   5
6    7    8
```

this function should return the following values in a patched mutter assuming you are in workspace#4
```
global.workspace_manager.get_active_workspace().get_neighbor(Meta.MotionDirection.UP).index()			//expected 1
global.workspace_manager.get_active_workspace().get_neighbor(Meta.MotionDirection.RIGHT).index()		//expected 5
global.workspace_manager.get_active_workspace().get_neighbor(Meta.MotionDirection.DOWN).index()		//expected 7
global.workspace_manager.get_active_workspace().get_neighbor(Meta.MotionDirection.LEFT).index()		//expected 3
```


here's the output of `mutter-3.30.2`
```
global.workspace_manager.get_active_workspace().get_neighbor(Meta.MotionDirection.UP).index()			//actual 3
global.workspace_manager.get_active_workspace().get_neighbor(Meta.MotionDirection.RIGHT).index()		//actual 4
global.workspace_manager.get_active_workspace().get_neighbor(Meta.MotionDirection.DOWN).index()		//actual 5
global.workspace_manager.get_active_workspace().get_neighbor(Meta.MotionDirection.LEFT).index()		//actual 4
```

it basically discards the possibility of moving workspaces to the right/left direction, probably doesn't consider layout (horizontal) in workspaces at all, which is the bug.
the keybindings actually work, if you press them you will see the popup, but the workspace won't be moved, because it's going to the same workspace in the right/left keybindings.


this is a mutter bug, I think we can do nothing about it. I tried the patch in this [article](https://medium.com/@hkdb/patch-gnome-3-30-for-workspace-grid-1ee8b1dd92f9), and everything worked well.

tested on Debian 10 buster.
